packages/cli-repl/src/logger.ts

```diff
@@ -1,53 +1,189 @@
 import redactInfo from 'mongodb-redact';
+import Analytics from 'analytics-node';
 import redactPwd from './redact-pwd';
 import { ObjectId } from 'bson';
 import pino from 'pino';
-import clr from './clr';
 import path from 'path';
 import os from 'os';
 
-function logger(bus: any, logDir: string) {
+interface ApiEventArguments {
+  pipeline?: any[];
+  query?: object;
+  options?: object;
+  filter?: object; 
+}
+
+interface ApiEvent {
+  method?: string; 
+  class?: string;
+  db?: string;
+  coll?: string;
+  arguments?: ApiEventArguments;
+}
+
+interface UseEvent {
+  db: string;
+}
+
+interface AsyncRewriterEvent {
+  original: string;
+  rewritten: string;
+}
+
+interface ShowEvent {
+  method: string;
+}
+
+interface ErrorEvent {
+  error: object;
+}
+
+interface ConnectEvent {
+  uri: string;
+  isAtlas: boolean;
+  isLocalhost: boolean;
+  serverVersion: string;
+  isEnterprise: boolean;
+  authType: string;
+  isDataLake: boolean;
+  dlVersion?: string;
+  isGenuine: boolean;
+  serverName: string;
+}
+
+export default function logger(bus: any, logDir: string) {
   const sessionID = new ObjectId(Date.now());
   const logDest = path.join(logDir, `${sessionID}_log`);
   const log = pino({ name: 'monogsh' }, pino.destination(logDest));
   console.log(`Current sessionID: ${sessionID}`);
+  let userId;
+  let telemetry;
+
+  let analytics = new NoopAnalytics();
+  try {
+    // this file gets written as a part of a release
+    console.log(require('./config.js'))
+    log.warn(require('./config.js').SEGMENT_API_KEY)
```

Comment:
if this whole file is only used for Analytics id give it a more specific name like, `analytics-config.js`